docs/todo-due-dates.md

```diff
@@ -0,0 +1,178 @@
+```
+
+**Via `ember-template-lint` CLI:**
+
+```
+# to configure todos to transition to errors 20 days after creation
+yarn ember-template-lint . --update-todo --days-to-error-todo 20
+```
+
+### Todos that are past-due become warnings, then errors
+
+Use case: give 20 days for todos to be considered `warning`s, then 5 additional days until they are considered `error`s.
+
+**Via config:**
+
+```json
+{
+  "lint-todo": {
+    "daysToWarn": 20,
+    "daysToError": 25
+  }
+}
+```
+
+**Via `eslint` CLI:**
+
+```
+# to configure todos to transition to warnings 20 days after creation and to errors 25 days after creation
+UPDATE_TODO=1 DAYS_TO_WARN_TODO=20 DAYS_TO_ERROR_TODO=25 yarn eslint . --format eslint-formatter-todo
+```
+
+**Via `ember-template-lint` CLI:**
+
+```
+# to configure todos to transition to warnings 20 days after creation and to errors 25 days after creation
+yarn ember-template-lint . --update-todo --days-to-warn-todo 20 --days-to-error-todo 25
+```
+
+### Invalid option: only `daysToWarn` is present
+
+If `daysToWarn` is passed without `daysToError` an error must be produced.
+
+**Via `eslint` CLI:**
+
+```
+$ UPDATE_TODO=1 DAYS_TO_WARN_TODO=20 yarn eslint . --format eslint-formatter-todo
+
+Error: `DAYS_TO_WARN_TODO` must be used with `DAYS_TO_ERROR_TODO`.
+```
+
+**Via `ember-template-lint` CLI:**
+
+```
+$ yarn ember-template-lint . --update-todo --days-to-warn-todo 20
+
+Error: `--days-to-warn-todo` must be used with `--days-to-error-todo`.
+```
+
+#### Invalid option: `daysToWarn` is greater or equal than `daysToError`
+
+If `daysToWarn` is greater or equal than `daysToError` an error must be produced:
+
+**Via `eslint` CLI:**
+
+```
+$ UPDATE_TODO=1 DAYS_TO_WARN_TODO=20 DAYS_TO_ERROR_TODO=5 yarn eslint . --format eslint-formatter-todo
+
+Error: `DAYS_TO_WARN_TODO` value must be less than `DAYS_TO_ERROR_TODO`.
+```
+
+**Via `ember-template-lint` CLI:**
+
+```
+$ yarn ember-template-lint . --update-todo --days-to-warn-todo 20 --days-to-error-todo=5
+
+Error: `--days-to-warn-todo` value must be less than `--days-to-error-todo`.
+```
+
+
+## Proposed changes to the schema to include due date
+
+```ts
+interface TodoData {
+  ...
+  created: Date;
+  daysToWarn?: int;
+  daysToError?: int;
+}
+```
+
+# Critique
+
+[critique]: #critique
+
+A few other options were considered:
+
+- Instead of due days, another option considered was passing a specific due date. Passing days is preferable because if we want to expose a config/CLI param for this, passing a date would make the config or command to run stale.
+- Allow the option to pass rule id with due date associated to them. The extra rule id config is not necessary since both `ember-template-lint` and `eslint` already have options to filter by rule id.
+- If devs needs extra functionality, we could provide a different npx tool to do that. For now, the goal is to keep the todo functionality simple with a simple due date mechanism.
```

Comment:
`npx` doesn't really have anything to do with this. It's rather that we could provide another CLI tool. `npx` is just a way we'd invoke it without installing the package.